### PR TITLE
#73 chatpage fix

### PR DESCRIPTION
**Pull request by Jordan141**
**Reference to related ticket:**
Reolves #72 , #73 

**Description of changes:**
- Correct homepath to solve component overlapping bug.
- Added missing return statement to generateRooms to show room list
- Bound this to socket callbacks

**Requirements**
Please use white box testing methods during review to ensure stability of branch

### CODE REVIEW

**Comment by Jordan141** [comment]
```
console.error node_modules\warning\warning.js:51
      Warning: You tried to redirect to the same route you're currently on: "/"
    console.log src\.tests\components\chatpage.test.js:53
```
Only occurs during test, will fix tomorrow. Adding WIP label to PR

**Comment by michaelKurowski** [comment]
@Jordan141 Ping me once CI will be fixed

File: src/client/src/.tests/components/chatpage.test.js [hunk 1]
```diff
@@ -117,13 +122,13 @@ describe('<ChatPage />', () => {
 	})
 	describe('<ChatPage /> Functionality Tests', () => {
 		it('Should update joined rooms', () => {
-			const EXPECTED_DATA = {roomId: DUMMY_ROOM, usernames: DUMMY_USER}
+			const EXPECTED_DATA = {roomId: DUMMY_ROOM, username: DUMMY_USER}
 			suite.Component.instance().updateJoinedRooms(EXPECTED_DATA)
 			const USERROOM_ROOMID = suite.Component.state(USERROOMS)[FIRST_INDEX].roomId
 			const USERROOM_USERNAMES = suite.Component.state(USERROOMS)[FIRST_INDEX].usernames
```
Comment (michaelKurowski) [inline src/client/src/.tests/components/chatpage.test.js]: Do we want to use `usernames` here, or `username`?

File: src/client/src/.tests/components/chatpage.test.js [hunk 2]
```diff
@@ -203,13 +208,13 @@ describe('<ChatPage />', () => {
 			expect(RETURN_VALUE).toBeUndefined()
 		})
 		it('Should handle the join event and update state', () => {
-			const JOIN_DATA = {roomId: DUMMY_ROOM, usernames: DUMMY_USER}
+			const JOIN_DATA = {roomId: DUMMY_ROOM, username: DUMMY_USER}
 			const RETURN_VALUE = suite.Component.instance().handleJoinEvent(JOIN_DATA)
 			const USERROOM_ROOMID = suite.Component.state(USERROOMS)[FIRST_INDEX].roomId
 			const USERROOM_USERNAMES = suite.Component.state(USERROOMS)[FIRST_INDEX].usernames
```
Comment (michaelKurowski) [inline src/client/src/.tests/components/chatpage.test.js]: Do we want to use `usernames` here, or `username`?

File: src/client/src/App.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const Register = require('./components/register')
 const ChatPage = require('./components/chatpage')
 const {BrowserRouter, Route} = require('react-router-dom')
 const {paths} = require('./routes/routes')
-const LOGIN_URL = paths.LOGIN
+const LOGIN_URL = paths.HOME
```
Comment (michaelKurowski) [inline src/client/src/App.js]: `conts loginUrl = paths.HOME`

File: src/client/src/components/chatpage.js [hunk 4]
```diff
@@ -12,20 +12,24 @@ const USERNAMES_PLACEHOLDER = ''
...
 	constructor(props) {
 		super(props)
+		const initProps = this.props.location.state
 		this.state = {
 			connected: false,
 			input: '',
 			messages: [],
 			selectedRoom: '',
-			username: this.props.location.state.username,
+			username: _.has(initProps, 'username') ? initProps['username'] : null,
```
Comment (michaelKurowski) [inline src/client/src/components/chatpage.js]: `_.get(initProps, 'username', null)`?

File: src/client/src/.tests/components/chatpage.test.js (hunk 1 above)
Comment (Jordan141) [reply src/client/src/.tests/components/chatpage.test.js]: These tests will be updated when I add functionality for multiple users of a single room.

File: src/client/src/.tests/components/chatpage.test.js (hunk 2 above)
Comment (Jordan141) [reply src/client/src/.tests/components/chatpage.test.js]: These tests will be updated when I add functionality for multiple users of a single room.

File: src/client/src/components/chatpage.js (hunk 4 above)
Comment (Jordan141) [reply src/client/src/components/chatpage.js]: Searches initialprops sent to the component for a username variable, if none is found, sets it to null, triggering redirect to login

**Pull request merged**